src/Microsoft.FeatureManagement/FeatureManagerSnapshot.cs

```diff
@@ -41,36 +42,18 @@ await foreach (string featureName in _featureManager.GetFeatureNamesAsync().Conf
             }
         }
 
-        public async Task<bool> IsEnabledAsync(string feature)
+        public Task<bool> IsEnabledAsync(string feature)
         {
-            //
-            // First, check local cache
-            if (_flagCache.ContainsKey(feature))
-            {
-                return _flagCache[feature];
-            }
-
-            bool enabled = await _featureManager.IsEnabledAsync(feature).ConfigureAwait(false);
-
-            _flagCache[feature] = enabled;
-
-            return enabled;
+            return _flagCache.GetOrAdd(
+                feature,
+                (key) => _featureManager.IsEnabledAsync(key));
```

Comment:
Should we have `.ConfigureAwait(false)` here?

Reply:
That's only needed where the task is awaited. So the consumer of this should `ConfigureAwait` if they are awaiting the result.

Reply:
Makes sense. We can't save the `ConfigureAwait` from the user code in this case.